app/src/main/java/org/mozilla/fenix/components/BackgroundServices.kt

```diff
@@ -76,35 +82,17 @@ class BackgroundServices(
         capabilities = setOf(DeviceCapability.SEND_TAB)
     )
     // If sync has been turned off on the server then disable syncing.
-    private val syncConfig = if (context.isInExperiment(Experiments.asFeatureSyncDisabled)) {
+    @VisibleForTesting(otherwise = PRIVATE)
+    val syncConfig = if (context.isInExperiment(Experiments.asFeatureSyncDisabled)) {
         null
     } else {
         SyncConfig(setOf(SyncEngine.HISTORY, SyncEngine.BOOKMARKS), syncPeriodInMinutes = 240L) // four hours
     }
 
-    val pushConfig by lazy {
-        val logger = Logger("PushConfig")
-        val projectIdKey = context.getString(R.string.pref_key_push_project_id)
-        val resId = context.resources.getIdentifier(projectIdKey, "string", context.packageName)
-        if (resId == 0) {
-            logger.warn("No firebase configuration found; cannot support push service.")
-            return@lazy null
-        }
+    val pushConfig by lazy { makePushConfig() }
+    private val pushService by lazy { FirebasePush() }
 
-        logger.debug("Creating push configuration for autopush.")
-        val projectId = context.resources.getString(resId)
-        PushConfig(projectId)
-    }
-
-    val pushService by lazy { FirebasePush() }
-
-    val push by lazy {
-        AutoPushFeature(
-            context = context,
-            service = pushService,
-            config = pushConfig!!
-        )
-    }
+    val push by lazy { makePush() }
```

Comment:
if push vars are not setup, the app crashes on startup

Reply:
Fixed